### ISSUE TITLE
PCap Playback Falis - Unable to Open Required Interface

### ISSUE THREAD

**Issue by Ben-Rosenfeld-VPS** [docsteer/sacnview#330]
Transmitting through the NIC works with no issues. However, the moment I try to use PCap Playback an error occurs which states "Unable to open required interface". The same error occurs whether we use pcap or pcapng files.

![image](https://github.com/docsteer/sacnview/assets/51710308/6d015741-a58f-46c1-95cc-947f4cf2bfd0)

Interface settings appear proper:

![image](https://github.com/docsteer/sacnview/assets/51710308/5b79656f-b173-4d2f-aeea-b1fc57e4bd49)

Latest beta build:

![image](https://github.com/docsteer/sacnview/assets/51710308/b081e2b1-ce1f-48a0-8566-e2053439158f)

**Comment by Ben-Rosenfeld-VPS** [comment]
Forgot to include image of NIC settings:
![image](https://github.com/docsteer/sacnview/assets/51710308/be897ba1-0b48-4084-9907-8beb3e80e1c9)

**Comment by marcusbirkin** [comment]
You'll need to run sACNView as Administrator (right click on shortcut, run-as administrator) to use this feature

**Comment by Ben-Rosenfeld-VPS** [comment]
Correct. Am already doing that as well. Is there something else that we are missing?

**Comment by marcusbirkin** [comment]
Do you have any additional network interfaces, other than the 1 listed in sACNView's preferences?

**Comment by Ben-Rosenfeld-VPS** [comment]
Yes. I have several network interfaces. I have tried three of them (all without luck):
1. Physical Ethernet Port on Laptop
2. Ethernet Port on USB C Dock
3. UGREEN USB C to Ethernet Adaptor

**Comment by marcusbirkin** [comment]
I mean, does Windows report any other network interfaces in Control Panel, other than the 1 listed in sACNView?
i.e. is sACNView missing adapters from it's list?

**Comment by Ben-Rosenfeld-VPS** [comment]
![image](https://github.com/docsteer/sacnview/assets/51710308/39e53b5e-3986-459b-be53-a464d9a89535)

vs

![image](https://github.com/docsteer/sacnview/assets/51710308/0826db20-6c8a-482c-a36b-6c7d4eedba38)

**Comment by marcusbirkin** [comment]
So yes, you have multiple adapters that don't appear in sACNView.
Also confusingly your "Ethernet 2", which was shown in the first post, is now shown in Windows control panel; but not in your most recent sACNView shot.

I would recommend that you do the following:
- Ensure Npacp is fully up to date, at time of post that appears to be v1.79
- Ensure that the system does not have WinPcap installed (the sACNView screenshot seems to show it is installed), as this does not work with Windows 11, and has poor performance with Windows 10.
- In Windows control panel, disable any and all interfaces you are not wanting to use for PCap playback (including WiFi, Bluetooth, virtual Ethernet).
- Ensure that the single remaining interface has only a single assigned IPv4 address (hint Windows hides the additional IP addresses behind an advanced button in the properties dialog).

**Comment by Ben-Rosenfeld-VPS** [comment]
Ok - I will try this and get back to you. Thanks for the recommendation.

**Comment by Ben-Rosenfeld-VPS** [comment]
Ok - sorry for the delay. 

Disabled all the I/Fs:
![image](https://github.com/docsteer/sacnview/assets/51710308/894b4613-3b51-46de-9f7d-4724745b8e84)

Updated Npacp to v1.79. Did not see WinPcap installed to remove from add/remove programs.

Ran sACN view as Admin but same issue:

![image](https://github.com/docsteer/sacnview/assets/51710308/6ce1e5ba-2226-4e68-a15a-8a595e213edf)

![image](https://github.com/docsteer/sacnview/assets/51710308/c3566748-0664-4dea-b5c8-705b053116d1)

**Comment by marcusbirkin** [comment]
...and you checked that the interface has only a single IP?
As for Wipcap, search your system folders for `wpcap.dll` and `packet.dll`, checking the version properties.

If you still have issues than there isn't much else that can be done, you might simply have to accept that this feature won't work on your machine.

**Comment by Ben-Rosenfeld-VPS** [comment]
Correct - the I/F only has a single IP.

The only version of wpcap.dll and packet.dll are the ones within the Npcap folder (that were installed with Npcap):
![image](https://github.com/docsteer/sacnview/assets/51710308/4017eab1-b277-4d03-9e99-776edd1fd48b)

Thanks for trying to support. As a side note - there are three other folks on my team who have this same issue. So I do not believe it is a singular event. This is likely a systemic issue.

**Comment by marcusbirkin** [comment]
You might try replacing `wpcap.dll` and `packet.dll` in the sACNView install folder with the dlls from `c:\Windows\Sytem32\Npcap`

**Comment by alphajbravo** [comment]
I just ran into this issue as well.  The only place I've gotten this to work at all is on a Win10 VM using a vmxnet3 adapter (vmware).  Using a USB NIC on the same vm (tried two different brands) I get the same error as the OP.  I also get the same error on two different Win11 laptops and two different Win10 desktops regardless with both USB and built-in PCIe adapters.

> You might try replacing `wpcap.dll` and `packet.dll` in the sACNView install folder with the dlls from `c:\Windows\Sytem32\Npcap`

I tried this, but I get the error when opening a .pcapng file, whether I use the v1.78 files from my current wireshark install or the latest v1.79

> Error opening <file> 
> bad dump file format

**Comment by marcusbirkin** [comment]
Do you have the same issue with the older .pcap formatted files?

After copying the DLLs, could you post a screenshot of the about screen from sACNView.

**Comment by alphajbravo** [comment]
After copying the npcap dlls, I can open and play back pcaps on the vmxnet3 adapter, but get the same 'Unable to open required interface' error on a USB adapter.  So behavior seems to be the same as the included dlls.  

I had been using v2.1.3, but just tried v2.2.0-beta.1, and it actually DOES work with the replaced DLLs on both the vmxnet3 and USB adapters on the Win10 VM, and with both pcap and pcapng files.  Screenshots of the about screens after replacing the dlls below.  With the included dlls, it seems to behave the same as v2.1.3

![image](https://github.com/docsteer/sacnview/assets/8129563/4caea3b5-e67c-4cc2-8300-a235d8c02a8c)
![image](https://github.com/docsteer/sacnview/assets/8129563/5c373e0a-7f6c-4406-ba6e-6b69a7e2300b)

**Comment by marcusbirkin** [comment]
So yes, this confirms what the issue is.
WinPcap fails on modern Windows.

sACNView is bundled with WinPcap (as it was current at the time), but can not be bundled with Npcap due to licence restrictions.
#331 removes the bundled WinPcap DLLs, and is will prefer Npcap if both are installed on the system.

In the meantime the solution is to ensure that only Npacp DLLs are available in accessible PATHs.

**Comment by alphajbravo** [comment]
I just tried the same solution on a Win11 laptop, and with v2.2.0-beta.1 and Npcap v1.79, and it works on a USB network adapter. If I try to use the built-in wifi, I get a different problem: as soon as I click "Play", I get what looks like an error popup with no text, and then sACNView closes.  I don't really WANT to send sACN over wifi, but I don't know if the underlying problem may occur with other adapters as well.

> In the meantime the solution is to ensure that only Npacp DLLs are available in accessible PATHs, as currently sACNView will use the first set of DLLs found.

At least with v2.2.0-beta.1, sACNView does not appear to locate the npcap dlls unless "Install Npcap in WinPcap API-compatible mode" option is selected during Npcap installation.  So it may be worth documenting that for users, or looking for the default Npcap path explicitly.

Thanks for the quick support on this.  pcap playback is a really powerful feature, and even with the WinPcap issues it saved me a ton of time in pinpointing the source of a challenging problem yesterday.

**Comment by marcusbirkin** [comment]
`Install Npcap in WinPcap API-compatible Mode.` places the DLLs in the system32 folder and not in the subdirectory system32/npcap.
The system32 folder is one of Window's search paths.
However the directory local to the executable is the first place checked - unless the executables working directory is altered (i.e. with a shortcut, or other method).

With #331, the npcap subfolder is set as a preferred search path. But ultimately it will depend on what ever the system provides.
